e2e-tests/custom-users-roles-sharded/run

```diff
@@ -0,0 +1,357 @@
+#!/bin/bash
+
+set -o errexit
+
+compare() {
+	local database="$1"
+	local command="$2"
+	local uri="$3"
+	local target="$4"
+
+	run_mongos "use ${database}\n ${command}" "$uri" "mongodb" \
+		| egrep -v 'I NETWORK|W NETWORK|F NETWORK|Error saving history file|Percona Server for MongoDB|connecting to:|Unable to reach primary for set|Implicit session:|versions do not match|Error saving history file:' \
+		| $sed -re 's/ObjectId\("[0-9a-f]+"\)//; s/-[0-9]+.svc/-xxxesvc/' \
+		| sed '/"userId"/d' \
+			>$tmp_dir/${target}
+
+	diff ${test_dir}/compare/${target}.json $tmp_dir/${target}
+}
+
+check_auth() {
+	local uri="$1"
+
+	ping=$(run_mongos "db.runCommand({ ping: 1 }).ok" "$uri" "" "" "--quiet" | egrep -v 'I NETWORK|W NETWORK|Error saving history file|Percona Server for MongoDB|connecting to:|Unable to reach primary for set|Implicit session:|versions do not match|Error saving history file:')
+	desc "ping return"
+	if [ "${ping}" != "1" ]; then
+		return 1
+	fi
+}
+
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+set_debug
+
+cluster="some-name"
+
+create_infra "$namespace"
+
+mongosUri="userAdmin:userAdmin123456@$cluster-mongos.$namespace"
+
+desc 'create secrets and start client'
+kubectl_bin apply -f "${conf_dir}/client.yml" \
+	-f "${conf_dir}/secrets.yml" \
+	-f "${test_dir}/conf/app-user-secrets.yml"
+
+
+apply_s3_storage_secrets
+if version_gt "1.19" && [ $EKS -ne 1 ]; then
+	cat "$conf_dir/container-rc.yaml" | $sed 's/docker/runc/g' | kubectl_bin apply -f -
+elif version_gt "1.24" && [ $EKS -eq 1 ]; then
+	cat "$conf_dir/container-rc.yaml" | $sed 's/docker/runc/g' | kubectl_bin apply -f -
+else
+	kubectl_bin apply -f "$conf_dir/container-rc.yaml"
+fi
+
+desc 'create first PSMDB cluster'
+
+apply_cluster "$test_dir/conf/$cluster-rs0.yml"
+
+desc 'check if all 3 Pods started'
+wait_for_running $cluster-rs0 3
+wait_for_running $cluster-cfg 3 "false"
+wait_for_running $cluster-mongos 3
+wait_cluster_consistency "${cluster}"
+
+desc 'check if service and statefulset created with expected config'
+compare_kubectl statefulset/$cluster-rs0
+compare_kubectl statefulset/$cluster-cfg
+compare_kubectl statefulset/$cluster-mongos ""
+
+desc 'check user created on cluster creation'
+
+userOne="user-one"
+userOnePass=$(getSecretData "user-one" "userOnePassKey")
+compare 'admin' 'db.getUser("user-one")' "$mongosUri" "user-one"
+check_auth "$userOne:$userOnePass@$cluster-mongos.$namespace"
+
+desc 'delete initial user from CR and create a new one'
+kubectl_bin patch psmdb ${cluster} --type=merge --patch '{
+		"spec": {"users":[
+			{
+				"name":"user-two",
+				"db":"admin",
```

Comment:
We always use admin DB why... Please use a custom one.

Reply:
There is:
```
desc 'check new user created after updated user db via CR'
kubectl_bin patch psmdb ${psmdb} --type=merge --patch '{
		"spec": {"users":[
			{
				"name":"user-three",
				"db":"newDb",
				"passwordSecretRef": {
					"name": "user-two",
					"key": "userTwoPassKey"
				},
				"roles": [
					{"db":"admin","name":"clusterAdmin"}
				]
			}
		]}
	}'
```